### PR TITLE
Turn on noUnusedLocals and noUnusedParameters

### PR DESCRIPTION
**Pull request by glasser**

### CODE REVIEW

File: packages/apollo-server-core/src/ApolloServer.ts [hunk 1]
```diff
@@ -374,7 +373,14 @@ export class ApolloServerBase<Request = RequestInit> {
       schema: this.schema,
       extensions: this.extensions,
       context,
-      // allow overrides from options
+      // Allow overrides from options. Be explicit about a couple of them to
```
Comment (glasser) [inline packages/apollo-server-core/src/ApolloServer.ts]: This is weird but I dunno if there's anything better.

File: packages/apollo-server-hapi/src/ApolloServer.ts [hunk 2]
```diff
@@ -33,10 +32,9 @@ export interface HapiListenOptions {
   launcherOptions?: EngineLauncherOptions;
 }
 
-const handleFileUploads = (
-  uploadsConfig: Record<string, any>,
-  server: ApolloServerBase<hapi.Request>,
-) => async (req: hapi.Request, h: hapi.ResponseToolkit) => {
+const handleFileUploads = (uploadsConfig: Record<string, any>) => async (
```
Comment (glasser) [inline packages/apollo-server-hapi/src/ApolloServer.ts]: These changes are large enough I want to make sure it's not a bug that server wasn't used

Comment (evans) [reply packages/apollo-server-hapi/src/ApolloServer.ts]: server used to be used when the configuration was in the server constructor. Now that it's configured in the `registerServer` call, we con't have to worry about it

File: packages/apollo-server-core/src/ApolloServer.ts (hunk 1 above)
Comment (evans) [reply packages/apollo-server-core/src/ApolloServer.ts]: Seems odd, though I don't think it's worth destructuring to assign the two variables [here](https://github.com/apollographql/apollo-server/pull/1126/files#diff-3467ccc78fbd796513acd747ff2110a5L85)

**Pull request merged**